lib/ensure/symlink.js

```diff
@@ -1,65 +1,50 @@
 'use strict'
 
-const u = require('universalify').fromCallback
+const u = require('universalify').fromPromise
 const path = require('path')
 const fs = require('../fs')
-const _mkdirs = require('../mkdirs')
-const mkdirs = _mkdirs.mkdirs
-const mkdirsSync = _mkdirs.mkdirsSync
 
-const _symlinkPaths = require('./symlink-paths')
-const symlinkPaths = _symlinkPaths.symlinkPaths
-const symlinkPathsSync = _symlinkPaths.symlinkPathsSync
+const { mkdirs, mkdirsSync } = require('../mkdirs')
 
-const _symlinkType = require('./symlink-type')
-const symlinkType = _symlinkType.symlinkType
-const symlinkTypeSync = _symlinkType.symlinkTypeSync
+const { symlinkPaths, symlinkPathsSync } = require('./symlink-paths')
+const { symlinkType, symlinkTypeSync } = require('./symlink-type')
 
-const pathExists = require('../path-exists').pathExists
+const { pathExists } = require('../path-exists')
 
 const { areIdentical } = require('../util/stat')
 
-function createSymlink (srcpath, dstpath, type, callback) {
-  callback = (typeof type === 'function') ? type : callback
-  type = (typeof type === 'function') ? false : type
+async function createSymlink (srcpath, dstpath, type = false) {
```

Comment:
Not sure why the old logic worked this way, but no reason to default to `false` here.
```suggestion
async function createSymlink (srcpath, dstpath, type) {
```

Reply:
Fixed in [`fbfc9c7` (#1023)](https://github.com/jprichardson/node-fs-extra/pull/1023/commits/fbfc9c7d86925105cf905df9e11b51132be386d4).